lib/models/src/main/java/org/sagebionetworks/repo/model/TeamState.java

```diff
@@ -0,0 +1,35 @@
+package org.sagebionetworks.repo.model;
+
+public enum TeamState {
+    // A team is open when a user can create a membership request for the team
+    OPEN,
+    // A team is public when a user can join without creating a membership request
+    PUBLIC,
+    // A team is closed when a user cannot create a membership request
+    CLOSED;
+
+    public static TeamState from(Team team) {
+        boolean canPublicJoin = Boolean.TRUE.equals(team.getCanPublicJoin());
+        boolean canRequestMembership;
+        if (canPublicJoin) {
```

Comment:
I would add some comments why we do this (e.g. to maintain backward compatibility with existing clients etc)